Off-Top-App/off-top-flutter
# OT-Authentication Service | Implemented Google Sign-In and Firebase Authentication Services

**What I Did:**

- Implemented communication between Firebase and user devices.

- Added functions for user to sign in using a Google Account.

- Redesigned login.dart to fit Google Sign In button.

- Add Loading Progress Indicator to wait for communication with Google Services and Firebase.

- Created a dart file for Google Sign In functions and placed login.dart and the Google Sign In file into a folder called login.

**How To Test:** 

- Prerequisites: You will need to run `flutter pub get` to pull the new packages for Google Sign In.

- You will need Zookeeper, Kafka, Spring-boot, and port forwarding running to log into app.

- **IMPORTANT** Android devices must add SHA1 certificate for their device/emulator into Firebase!

> 
> 1. In flutter terminal enter:
> 
> >Mac: `keytool -list -v -alias androiddebugkey -keystore ~/.android/debug.keystore`
> >Windows: `keytool -list -v -alias androiddebugkey -keystore C:\Users\USERFOLDER\.android\debug.keystore`   Replace USERFOLDER with your own folder.
> 
> 2. Password for both Mac/Windows is `android` .
>
> 3. Locate the SHA1 Certificate output from the command and copy the entire string.
> 
> 4. Open Firebase web console and locate Android App Settings. It can be located in the "2 apps" tab button. Credentials for Firebase login can be found in our [documentation](https://docs.google.com/document/d/1kqS7te_Rzmt-06OILNQuTDpwEqzwJCJ0VYcxnONbMy4/edit?usp=sharing).
>
> ><img width="462" alt="Screen Shot 2020-06-18 at 11 35 06 PM" src="https://user-images.githubusercontent.com/48227740/85103865-59654b00-b1bc-11ea-9465-32b5f8a6c0ce.png">
> 
> 5. Click on settings and locate SHA certificate fingerprints at the bottom of the page.
> 
> 6. Click on add Add Fingerprint and paste your SHA1 Certificate from step 3.

- **IMPORTANT** iOS devices must redirect their path for the GoogleService-Info.plist file using Xcode.

> 1. Locate your iOS folder within your off-top-flutter directory.

> 2. Right click the iOS directory inside your IDE(Android Studio or VS Code) and click "open in Xcode".

>3. In Xcode, Navigate to /Runner/Runner/GoogleService-Info.plist. 

><img width="277" alt="Screen Shot 2020-06-21 at 9 42 42 PM" src="https://user-images.githubusercontent.com/48227740/85249265-2616fc00-b408-11ea-9f54-279b9d253b97.png">

>4. On right hand, click on the file icon.
><img width="276" alt="Screen Shot 2020-06-21 at 9 43 54 PM" src="https://user-images.githubusercontent.com/48227740/85249371-6fffe200-b408-11ea-9b80-8786cb891fb7.png">

>5. Locate your "off-top-flutter" folder. Navigate to /iOS/Runner and locate the "GoogleService-Info.plist" file and select it. Afterwards, confirm choose.

>6. Close Xcode.

- Open MySQL and create a new entry of user containing the email you plan to test with.
>**NOTE** 
>You will need to enter in all data that pertains to the user I.E "username, email, first name, etc." This could fake information but the email used has to be real.

- Test the app by signing in with your Google Account by clicking the "Sign in With Google Button" on the Login Page. This will be the first screen prompted when running the app.

- Test on both iOS and Android.
